library/Zend/Http/Client/Adapter/Socket.php

```diff
@@ -238,7 +245,11 @@ public function connect($host, $port = 80, $secure = false)
 
             //distinguish between request timeout and connect timeout like in curl adapter
             // request_timeout defaults to connection timeout to keep backwards compatibility
-            if(!array_key_exists('request_timeout', $this->config)) {
```

Comment:
* the first time here, `timeout == 1` so we enter the IF and set `request_timeout = timeout = 1`.
* the second time here, `timeout == 117`, but since `request_timeout` is set, it remains at `request_timeout == 1` ... and we run out of time.